engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -85,79 +91,119 @@ class EngineJobExecutionActor(jobKey: BackendJobDescriptorKey,
   // When PreparingJob, the FSM always has NoData
   when(PreparingJob) {
     case Event(BackendJobPreparationSucceeded(jobDescriptor, bjeaProps), NoData) =>
+      val updatedData = NoResponseData(jobDescriptor, bjeaProps)
+      // If this backend doesn't offer a cache hit copying actor, remove 'read' from the effective cache options.
+      // i.e. there's no way to copy a cache hit so don't bother looking for cache hits.  If this backend does
+      // offer a cache hit copying actor, squirrel away its props so we can create this actor later if there's a hit.
+      if (effectiveCallCachingMode.readFromCache) {
+        factory.cacheHitCopyingActorProps(jobDescriptor, initializationData, serviceRegistryActor) match {
+          case Some(props) =>
+            callCachingActorProps = props
+          case None =>
+            log.warning("{}: Turning off cache reading for this job as no cache copying actor is available for this backend.", jobTag)
+            effectiveCallCachingMode = effectiveCallCachingMode.withoutRead
+        }
+      }
       effectiveCallCachingMode match {
         case activity: CallCachingActivity if activity.readFromCache =>
           initializeJobHashing(jobDescriptor, activity)
-          goto(CheckingCallCache) using JobDescriptorData(jobDescriptor, bjeaProps)
+          goto(CheckingCallCache) using updatedData
         case activity: CallCachingActivity =>
           initializeJobHashing(jobDescriptor, activity)
-          runJob(jobDescriptor, bjeaProps)
-        case CallCachingOff => runJob(jobDescriptor, bjeaProps)
+          runJob(updatedData)
+        case CallCachingOff => runJob(updatedData)
       }
     case Event(response: BackendJobPreparationFailed, NoData) =>
       context.parent forward response
       context stop self
       stay()
   }
 
-  // When CheckingCallCache, the FSM always has EJEAJobDescriptorData
+  private def addHashesAndStay(data: NoResponseData, hashes: CallCacheHashes): State = {
```

Comment:
ToL: I like to put the helper function below the state list, so the `when(state)`s are all listed together